packages/vite/src/node/plugins/define.ts

```diff
@@ -29,9 +29,13 @@ export function definePlugin(config: ResolvedConfig): Plugin {
       'import.meta.hot': `false`
     })
   }
-
   const replacements: Record<string, string | undefined> = {
-    'process.env.NODE_ENV': JSON.stringify(config.mode),
+    // Note it is possible for user to have a custom mode, e.g. `staging` where
+    // production-like behavior is expected. This is indicated by NODE_ENV=production
+    // loaded from `.staging.env` and set by us as VITE_USER_NODE_ENV
+    'process.env.NODE_ENV': JSON.stringify(
+      process.env.VITE_USER_NODE_ENV || config.mode
+    ),
```

Comment:
This should be `process.env.NODE_ENV || config.mode`.  https://github.com/vitejs/vite/blob/352cd397d8c9d2849690e3af0e84b00c6016b987/packages/vite/src/node/config.ts#L281-L285

Reply:
See the comments, both can work normally. I change it now

Reply:
done